pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/ingress_reconciler.go

```diff
@@ -78,9 +78,43 @@ func (ir *IngressReconciler) Reconcile(ctx context.Context, isvc *v1beta1.Infere
 	serviceHost := getServiceHost(isvc)
 	serviceUrl := getServiceUrl(isvc, ir.ingressConfig)
 	disableIstioVirtualHost := ir.ingressConfig.DisableIstioVirtualHost
+
+	// If the "serving.kserve.io/stop" annotation is present in isvc, follow the force stop logic
+	forceStopRuntime := "false"
+	if val, exist := isvc.Annotations[constants.StopAnnotationKey]; exist {
+		forceStopRuntime = val
+	}
+
+	if strings.EqualFold(forceStopRuntime, "true") {
+		// Delete the service
+		existingService := &corev1.Service{}
+		if err := ir.client.Get(ctx, types.NamespacedName{Name: isvc.Name, Namespace: isvc.Namespace}, existingService); err == nil {
+			log.Info("The InferenceService ", isvc.Name, " is marked as stopped — delete its associated Service")
+			if err := ir.client.Delete(ctx, existingService); err != nil {
+				return err
+			}
+		} else if !apierr.IsNotFound(err) {
+			return err
+		}
+
+		// Delete the virtualservice
+		existingVService := &istioclientv1beta1.VirtualService{}
+		if err := ir.client.Get(ctx, types.NamespacedName{Name: isvc.Name, Namespace: isvc.Namespace}, existingVService); err == nil {
+			log.Info("The InferenceService ", isvc.Name, " is marked as stopped — delete its associated VirtualService")
+			if err := ir.client.Delete(ctx, existingVService); err != nil {
```

Comment:
To prevent deleting a user-owned resource, I suggest validating the fecthed `VirtualService` is owned by the `InferenceService`.

This comment would also apply to the `Service`.

Reply:
is this needed? We are setting the owner as the inference service through SetControllerReference - I don't see how a virtual service could get made/updated and skip the SetControllerReference call